### PR TITLE
Add Carousel component to UI Kits

### PR DESCRIPTION
**Pull request by aminpaks**
## Carouse component
Built based on `react-slick` which is built based on `slick-carousel`

## How to test
Head over to styleguide and click on Carousel

### CODE REVIEW

File: src/Styleguide/Page.tsx [hunk 1]
```diff
@@ -1,34 +1,35 @@
...
         { title: 'Buttons', url: './buttons' },
+        { title: 'Carousels', url: './carousels' },
         { title: 'Grid System', url: './grid-system' },
         { title: 'Panels', url: './panels' },
         { title: 'Tabs', url: './tabs' },
         { title: 'Typographies', url: './typographies' },
       ]}
     >
       <LayoutSection>
+        {isStyleguideRoot && <Redirect to="/styleguide/buttons" />}
```
Comment (aminpaks) [inline src/Styleguide/Page.tsx]: Better than the old approach (removed line 17).

File: src/UI-Kits/Carousel/Carousel.styled.ts [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+      > div > div {
+        outline: none;
+        cursor: ${props => (props.draggable ? 'default' : undefined)};
+      }
+    }
+
+    .slick-arrow {
+      top: calc(
+        50% +
+          ${getPropValue({ prop: 'arrowSize', defaultValue: defaultArrowSize, normalize: 0.03 })}
```
Comment (aminpaks) [inline src/UI-Kits/Carousel/Carousel.styled.ts]: Look how simple is to calculate a value, get `arrowSize` property and set to a default value if not exists and return 3% of it.

File: src/UI-Kits/Carousel/Carousel.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+  const updateChildren = Children.map(children, child => {
+    if (isChildTypeOf(CarouselSlide, child)) {
+      // This is necessary to preserve the customized styling
+      return cloneElement(child, { __internalStyle: child.props.style, __internalProps: props });
+    }
+    return null;
+  });
+
+  return (
+    <StyledCarouselsContainer {...{ arrowSize, slidePadding, draggable: props.draggable }}>
```
Comment (aminpaks) [inline src/UI-Kits/Carousel/Carousel.tsx]: We should avoid passing event handlers for obvious reasons.

File: src/UI-Kits/Carousel/Carousel.styled.ts (hunk 2 above)
Comment (nasreddineskandrani) [reply src/UI-Kits/Carousel/Carousel.styled.ts]: humm not sure to understand all of it here.

File: src/UI-Kits/Carousel/Carousel.tsx (hunk 3 above)
Comment (nasreddineskandrani) [reply src/UI-Kits/Carousel/Carousel.tsx]: can we open an issue to fix it?

**Comment by nasreddineskandrani** [comment]
image

File: src/UI-Kits/Carousel/Carousel.tsx (hunk 3 above)
Comment (aminpaks) [reply src/UI-Kits/Carousel/Carousel.tsx]: Not an issue explained in the issue. A more solid solution would be writing a utility function to exclude the event handlers.

File: src/UI-Kits/Carousel/Carousel.styled.ts (hunk 2 above)
Comment (aminpaks) [reply src/UI-Kits/Carousel/Carousel.styled.ts]: Its ok.

**Pull request merged**